src/semver.nim

```diff
@@ -11,40 +11,53 @@ import std/strutils
 # it only handles basic dot-separated version format
 
 type Version* = ref object
-  major: int
-  minor: int
-  patch: int
-  name:  string
+  major:  int
+  minor:  int
+  patch:  int
+  suffix: string
+  name:   string
 
 proc parseVersion*(version: string): Version =
   var
-    major = 0
-    minor = 0
-    patch = 0
+    major  = 0
+    minor  = 0
+    patch  = 0
+    suffix = ""
   let
-    name  = version.strip(chars={'v', ',', '.'})
-    parts = name.split('.')
+    name     = version.strip(chars={'v'}, trailing=false).strip(chars={',', '.', '-', '+'})
```

Comment:
Maybe also want to strip out a capitalised 'V' prefix ?

Reply:
sure why not

Reply:
done